CMakeLists.txt

```diff
@@ -57,80 +37,112 @@ target_sources(
+          src/requesthandler/RequestHandler_Filters.cpp
+          src/requesthandler/RequestHandler_General.cpp
+          src/requesthandler/RequestHandler_Inputs.cpp
+          src/requesthandler/RequestHandler_MediaInputs.cpp
+          src/requesthandler/RequestHandler_Outputs.cpp
           src/requesthandler/RequestHandler_Sources.cpp
+          src/requesthandler/RequestHandler_Record.cpp
           src/requesthandler/RequestHandler_Scenes.cpp
-          src/requesthandler/RequestHandler_Inputs.cpp
-          src/requesthandler/RequestHandler_Transitions.cpp
-          src/requesthandler/RequestHandler_Filters.cpp
           src/requesthandler/RequestHandler_SceneItems.cpp
-          src/requesthandler/RequestHandler_Outputs.cpp
           src/requesthandler/RequestHandler_Stream.cpp
-          src/requesthandler/RequestHandler_Record.cpp
-          src/requesthandler/RequestHandler_MediaInputs.cpp
+          src/requesthandler/RequestHandler_Transitions.cpp
           src/requesthandler/RequestHandler_Ui.cpp
-          src/requesthandler/RequestHandler.h
-          src/requesthandler/RequestBatchHandler.cpp
-          src/requesthandler/RequestBatchHandler.h
           src/requesthandler/rpc/Request.cpp
           src/requesthandler/rpc/Request.h
           src/requesthandler/rpc/RequestBatchRequest.cpp
           src/requesthandler/rpc/RequestBatchRequest.h
           src/requesthandler/rpc/RequestResult.cpp
           src/requesthandler/rpc/RequestResult.h
           src/requesthandler/types/RequestStatus.h
-          src/requesthandler/types/RequestBatchExecutionType.h
+          src/requesthandler/types/RequestBatchExecutionType.h)
+
+target_sources(
+  obs-websocket
+  PRIVATE src/utils/Compat.cpp
+          src/utils/Compat.h
           src/utils/Crypto.cpp
           src/utils/Crypto.h
           src/utils/Json.cpp
           src/utils/Json.h
           src/utils/Obs.cpp
-          src/utils/Obs_StringHelper.cpp
-          src/utils/Obs_NumberHelper.cpp
+          src/utils/Obs.h
+          src/utils/Obs_ActionHelper.cpp
           src/utils/Obs_ArrayHelper.cpp
+          src/utils/Obs_NumberHelper.cpp
           src/utils/Obs_ObjectHelper.cpp
+          src/utils/Obs_StringHelper.cpp
           src/utils/Obs_SearchHelper.cpp
-          src/utils/Obs_ActionHelper.cpp
-          src/utils/Obs.h
           src/utils/Obs_VolumeMeter.cpp
           src/utils/Obs_VolumeMeter.h
           src/utils/Obs_VolumeMeter_Helpers.h
           src/utils/Platform.cpp
           src/utils/Platform.h
-          src/utils/Compat.cpp
-          src/utils/Compat.h
-          src/utils/Utils.h
-          deps/qr/cpp/QrCode.cpp
-          deps/qr/cpp/QrCode.hpp)
+          src/utils/Utils.h)
+
+target_sources(obs-websocket PRIVATE deps/qr/cpp/QrCode.cpp deps/qr/cpp/QrCode.hpp)
+
+configure_file(src/plugin-macros.h.in src/plugin-macros.generated.h)
+target_sources(obs-websocket PRIVATE src/plugin-macros.generated.h)
```

Comment:
The generated file should be `${CMAKE_CURRENT_SOURCE_DIR}/src/plugin-macros.generated.h`